src/main/java/com/squareup/osstrich/JavadocPublisher.java

```diff
@@ -157,21 +158,37 @@ private void writeIndexFiles(String groupId, Multimap<String, Artifact> artifact
     }
   }
 
-  private String findRelativePath(File directory, String artifactId) {
+  private @Nullable String findRelativePath(File directory, String artifactId) {
     if (!directory.isDirectory()) {
       throw new IllegalArgumentException(directory + " is not a directory!");
     }
+
+    // Look for Javadoc in the current directory.
     if (new File(directory, "index.html").exists()) {
       return directory.getName();
-    } else {
-      // Look for subdirectories of the same name as the artifact and go a level deeper
-      File possibleSubDir = new File(directory, artifactId);
-      if (possibleSubDir.isDirectory()) {
-        return directory.getName() + File.separator + findRelativePath(possibleSubDir, artifactId);
-      }
-      throw new RuntimeException("Could not find a valid indexed path for " + artifactId + ". Files are "
-              + Arrays.toString(Objects.requireNonNull(directory.listFiles())));
     }
+
+    // Look for subdirectories of the same name as the artifact and go a level deeper
+    File artifactDir = new File(directory, artifactId);
+    if (artifactDir.isDirectory()) {
+      String relativePath = findRelativePath(artifactDir, artifactId);
+      return relativePath != null
+          ? (directory.getName() + File.separator + relativePath)
+          : null;
+    }
+
+    // For Kotlin multiplatform look for a subdirectory named 'jvm' and go a level deeper.
+    File jvmDir = new File(directory, "jvm");
+    if (jvmDir.isDirectory()) {
+      String relativePath = findRelativePath(jvmDir, artifactId);
+      return relativePath != null
+          ? (directory.getName() + File.separator + relativePath)
+          : null;
+    }
+
+    log.error("Could not find a valid indexed path for " + artifactId
+        + ". Files are " + Arrays.toString(Objects.requireNonNull(directory.listFiles())));
```

Comment:
readability nit - static import the `requireNonNull`

Reply:
Fixed!